pkg/apis/core/helper/helpers.go

```diff
@@ -195,7 +195,7 @@ func TaintsAreTolerated(taints []core.SeedTaint, tolerations []core.Toleration)
 
 // SeedSettingExcessCapacityReservationEnabled returns true if the 'excess capacity reservation' setting is enabled.
 func SeedSettingExcessCapacityReservationEnabled(settings *core.SeedSettings) bool {
-	return settings == nil || settings.ExcessCapacityReservation == nil || settings.ExcessCapacityReservation.Enabled
+	return settings != nil && settings.ExcessCapacityReservation != nil && pointer.BoolDeref(settings.ExcessCapacityReservation.Enabled, true)
```

Comment:
Shouldn't it be

```suggestion
	return settings == nil || settings.ExcessCapacityReservation == nil || pointer.BoolDeref(settings.ExcessCapacityReservation.Enabled, true)
```

?

IIUC, we want to treat the field as enabled by default and only be disabled if it is explicitly set to false.

Otherwise with the current implementation:
- settings=nil -> seed setting disabled
- settings.excessCapacityReservation=nil -> seed setting disabled
- settings.excessCapacityReservation.enabled=nil -> seed setting enabled

which seems to be confusion/inconsistent?